search-sentiments-map/routes/country-trends.js

```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the 'License'); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an 'AS IS' BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+const express = require('express');
+const router = express.Router();  // Using Router to divide the app into modules.
+
+const {Datastore} = require('@google-cloud/datastore');
+const datastore = new Datastore();
+
+/** Renders the trends data of the country given by the request parameter. */
+router.get('/:country', (req, res) => {
+  var country = req.params.country;
+  retrieveCountryTrends(country).then(trends => {
+    res.setHeader('Content-Type', 'application/json');
+    res.send(trends);
+  });
+});
+
+/**
+ * Gets the trends data (each trend including title, articles, and sentiment) 
+ * from the datastore for the specified country.
+ */
+async function retrieveCountryTrends(country) {
+  const query = datastore.createQuery('TrendsEntry').order('timestamp', {
+    descending: true,
+  });
+  const [trendsEntry] = await datastore.runQuery(query);
+
+  const countryTrends = trendsEntry[0].trendsByCountry
+      .filter(trends => trends.country === country);  
```

Comment:
You may be able to filter through the query itself. The Datastore will handle the filtering and you won't need to do it here, which I believe I read cuts down on runtime.

Reply:
I don't know how you'd query this in Datastore, since it is store as a single entity not multiple entries. Can you send me the article where you read that?

Reply:
Oh I see it's one entity. Don't worry about it then.